### PR TITLE
Validate metric names

### LINKED ISSUE #184: Validate metric names and make them valid when possible
Some MIB files produce metric names that have dashes in them (like digital-sen1-1).

Should the snmp_exporter convert dashes to underscores to make these into valid metric names?

In this case, the problem metric is coming off of an older AVTECH TemPageR device. I'm currently building snmp_exporter from source (aa8c078) on a Linux machine.

### PR DESCRIPTION
**Pull request by Conorbro**
Fixes #184

### CODE REVIEW

File: generator/tree.go [hunk 1]
```diff
@@ -160,6 +160,10 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
 				return // Inaccessible metrics.
 			}
 
+			if strings.Contains(n.Label, "-") {
```
Comment (brian-brazil) [inline generator/tree.go]: You only need this change, plus a unittest for it.

**Comment by Conorbro** [comment]
@brian-brazil

File: generator/tree_test.go [hunk 2]
```diff
@@ -613,3 +613,44 @@ func TestGenerateConfigModule(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateMetricNames(t *testing.T) {
```
Comment (brian-brazil) [inline generator/tree_test.go]: This can go as part of TestGenerateConfigModule

**Comment by Conorbro** [comment]
@brian-brazil

File: generator/tree.go [hunk 3]
```diff
@@ -160,6 +160,10 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
...
+				n.Label = strings.Replace(n.Label, "-", "_", -1)
+			}
+
 			metric := &config.Metric{
 				Name:    n.Label,
```
Comment (brian-brazil) [inline generator/tree.go]: Replace it only down here, changing the actual n.Label could cause issues elsewhere.

**Comment by Conorbro** [comment]
@brian-brazil

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**